File: Basics of Input and Output/Number of steps.java
import java.io.*;
import java.util.*;

public class NumberOfSteps {
	static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
	static StringTokenizer st;
	public static void main(String[] args) throws IOException {
		int n = readInt();
		int[] a = new int[n], b = new int[n];
		for (int i = 0; i < n; i ++) a[i] = readInt();
		for (int i = 0; i < n; i ++) b[i] = readInt();
		
		int[] sum = new int[5001], freq = new int[5001];
		
		for (int i = 0; i < n; i ++) {
			int cnt = 0;
			for (int j = a[i]; j >= 0; j -= b[i]) { 
				freq[j] ++;
				sum[j] += cnt;
				cnt ++;
				if (b[i] == 0) break;
			} 
		}
		
		int ans = Integer.MAX_VALUE;
		for (int i = 0; i <= 5000; i ++) if (freq[i] == n) ans = Math.min(ans, sum[i]);
		System.out.println(ans == Integer.MAX_VALUE ? -1 : ans);
	}

	static String next() throws IOException {
		while (st == null || !st.hasMoreTokens()) 
			st = new StringTokenizer(br.readLine().trim());
		return st.nextToken();
	}
	static long readLong() throws IOException {
		return Long.parseLong(next());
	}
	static int readInt() throws IOException {
		return Integer.parseInt(next());
	}
	static double readDouble() throws IOException {
		return Double.parseDouble(next());
	}
	static char readCharacter() throws IOException {
		return next().charAt(0);
	}
	static String readLine() throws IOException {
		return br.readLine().trim();
	}
}

File: Basics of Input and Output/Number of steps.cpp
#include <iostream>
 
using namespace std;
 
int main() {
  int n;
  cin >> n;
  int a[n], b[n];
  for (int i = 0; i < n; i ++) cin >> a[i];
  for (int i = 0; i < n; i ++) cin >> b[i];

  int sum[5001] = {0}, freq[5001] = {0};
  
  for (int i = 0; i < n; i ++) {
    int cnt = 0;
      for (int j = a[i]; j >= 0; j -= b[i]) { 
      freq[j] ++;
      sum[j] += cnt;
      cnt ++;
      if (b[i] == 0) break;
    }
  }

  int ans = -1;
  for (int i = 0; i <= 5000; i ++) {
    if (freq[i] == n) if (sum[i] < ans || ans == -1) ans = sum[i];
  }
 
  cout << ans;
  return 0;
}
